### ISSUE TITLE
Overflow error when the returned value is an integer greater than Integer.MAX_VALUE or less than Integer.MIN_VALUE

### ISSUE THREAD

**Issue by matteozanon** [mrj0/jep#21]
Jep's getValue method always returns a java.lang.Integer if the python object is an integer.
This causes an overflow error when the value exceeds Integer.MAX_VALUE.

E.g.:
`jep.eval("x = int(1e10)")`
`jep.getValue("x") --> 1410065408 (java.lang.Integer)`

I think the error is in this statement in util.h file
`#define PyInt_Check(i)                    PyLong_Check(i)`

and in the pyembed.c file where the int check is performed (row 981).

In short, in pyembed.c a python integer is checked using PyInt_Check() which is just a redefine for PyLong_Check() and thus accepts also integer values that would need to be mapped into Java Long. Then, because the Python object was accepted by PyInt_Check(), the code returns a Java Integer.

**Comment by ndjensen** [comment]
You're using Python 3, correct?  Thanks for reporting, that's a good analysis.

**Comment by matteozanon** [comment]
Yes, I'm on Python3

**Comment by matteozanon** [comment]
I have the same problem with Float and Double too.
May I suggest to return always a Long or a Double? What do you think about it?

**Comment by ndjensen** [comment]
I pushed up a fix.  We have to retain support for 32-bit integers for projects using Python 2, since Python 2 still has a distinction between 32-bit and 64-bit ints/longs.  I made the change from floats to doubles as I don't see a reason to force precision loss in that method.  I would like to leave the ticket open for a bit so you can test it out, I can test it more, and to see if @mrj0 has any commentary.

**Comment by mrj0** [comment]
I don't remember why that was. Probably fine to change :-)

**Comment by ndjensen** [comment]
Closing as fixed in 3.4.
